src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,180 @@
+import java.util.Iterator;
 import java.util.List;
 import java.util.Optional;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 @Dao
 public class CarDaoImpl implements CarDao {
     @Override
     public Car create(Car car) {
-        Storage.addCar(car);
+        String queryForCar = "INSERT INTO cars (manufacturer_id, car_model) values (?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatementOfCar = connection.prepareStatement(queryForCar,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatementOfCar.setLong(1, car.getManufacturer().getId());
+            preparedStatementOfCar.setString(2, car.getModel());
+            preparedStatementOfCar.executeUpdate();
+
+            ResultSet generatedKeys = preparedStatementOfCar.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't create the data - " + car, exception);
+        }
         return car;
     }
 
     @Override
     public Optional<Car> get(Long id) {
-        return Storage.getCars().stream()
-                .filter(x -> x.getId().equals(id))
-                .findFirst();
+        Optional<Car> optionalCar;
+        String query = "select c.car_id as c_cid, c.manufacturer_id as c_mid, c.car_model as c_cm,"
+                + " d.driver_id as d_did, d.driver_name as d_dn, d.driver_license_number as d_dln"
+                + " from cars c"
+                + " inner join cars_drivers cd on c.car_id = cd.car_id"
+                + " inner join drivers d on d.driver_id = cd.driver_id"
+                + " where c.car_id = ? and c.car_deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            optionalCar = Optional.of(getCarFromDB(resultSet));
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get the data by id: " + id, exception);
+        }
+        return optionalCar;
     }
 
     @Override
     public List<Car> getAll() {
-        return Storage.getCars();
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT car_id from cars where car_deleted = false";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Optional<Car> optionalCar = get(resultSet.getObject("car_id", Long.class));
```

Comment:
here we are opening one connection inside of another - this is not good

let's look through the code and refactor the lines where we have this

Reply:
Can you help me? I don't know how I should refactor my code that won't break my logic

Reply:
I think my code is very bad and I am thinking to rewrite everything

Reply:
sure, we'll help